app/app/users/[user_id]/page.tsx

```diff
@@ -55,18 +55,51 @@ export default function UserDetails({
 }: {
   params: { user_id: string };
 }) {
+  const router = useRouter();
   const token = useToken();
   const userId = params.user_id;
   const { data: user, mutate } = useUser(userId, token);
-  const [showNewUserModal, setShowNewUserModal] = useState(false);
-  const onCloseModal = () => setShowNewUserModal(false);
+  const [showEditUserModal, setShowEditUserModal] = useState(false);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const onCloseModal = () => setShowEditUserModal(false);
 
   const onUpdateUserCallback = useCallback(async () => {
     // refetch the user details and close
     await mutate();
-    setShowNewUserModal(false);
+    setShowEditUserModal(false);
   }, [mutate]);
 
+  const onDeleteUser = useCallback(async () => {
+    setIsDeleting(true);
+    const url = `${
+      process.env.NEXT_PUBLIC_CR3_API_DOMAIN
+    }/user/delete_user/${encodeURIComponent(userId)}`;
+    const method = "DELETE";
+    try {
+      const response = await fetch(url, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+        method,
+      });
+      if (!response.ok) {
+        // the API may not return JSON for this endpoint — unclear from Auth0 docs
+        // auth0 api docs: https://auth0.com/docs/api/management/v2/users/delete-users-by-id
+        // very low priority todo: implement better error handling
+        const responseText = await response.text();
+        console.error(responseText);
+        window.alert(`Failed to delete user: ${String(responseText)}`);
+      } else {
+        router.push(`/users`);
+      }
+    } catch (err) {
+      console.error(err);
+      window.alert(`Failed to delete user: An unknown error has occured`);
+    }
+    setIsDeleting(false);
+  }, [router, token, userId]);
+
```

Comment:
i started to implement a full-blown delete modal, but the current VZE just uses `window.confirm` and considering how minor of a feature this is I didn't bother with a fancy modal 🚢